### PR TITLE
Testcontainers

### PR DESCRIPTION
**Pull request by ZombieAlienRobot**
## Description of changes
Added testcontainers for tests and improved the CI pipeline
  
## How has this been tested?
Please describe the test strategy you followed.

- [X] automated unit test
- [X] automated integration test
- [X] automated acceptance test
- [ ] manual, exploratory test

## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [X] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/test/resources/applicationt.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+spring.datasource.url=jdbc:tc:postgresql:///content-service
```
Comment (PaulBredl) [inline src/test/resources/applicationt.properties:2]: Typo in file name

File: src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java [hunk 2]
```diff
@@ -15,10 +19,14 @@
 
 @SpringBootTest(classes = ContentServiceApplication.class)
 @AutoConfigureGraphQlTester
+@ActiveProfiles("test")
 class ContentServiceApiTests {
     @Autowired
     private GraphQlTester graphQlTester;
 
+    @ClassRule
+    public static PostgreSQLContainer<GitsPostgresSqlContainer> postgreSQLContainer = GitsPostgresSqlContainer.getInstance();
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java:28]: @ClassRule is a Junit4 concept and were using Junit 5. Does the library not support Junit 5? Should usually be done with @ExtendWith

Comment (PaulBredl) [reply src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java:28]: Okay I think it should be done with @Container, see https://java.testcontainers.org/test_framework_integration/junit_5/

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java (hunk 2 above)
Comment (ZombieAlienRobot) [reply src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java:30]: That is correct changed it

File: src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java [hunk 3]
```diff
@@ -20,11 +21,12 @@
 @SpringBootTest(classes = ContentServiceApplication.class)
 @AutoConfigureGraphQlTester
 @ActiveProfiles("test")
+@Testcontainers
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/ContentServiceApiTests.java:24]: Should we put that in the "GraphQlApi" annotation maybe? No need to do it now, but in the future

**Review by PaulBredl — APPROVED** [review]
